### PR TITLE
Actions: Generate and Compile CLs without needing a webhook!™

### PR DESCRIPTION
**Pull request by Crossedfall**
## About The Pull Request

Having a PHP based webhook locked away in your mothers basement is a terrible way to generate CLs. Using our patented `A C T I O N` technology, we can generate and compile our CLs directly in GitHub! 

CLs won't be compiled after every push anymore. Instead they'll compile once every 15 minutes.

## Why It's Good For The Game

It makes CL handling a fair bit cleaner and removes an external dependency. 

## Changelog
:cl:
code: Nothing really changed. Just adding this here as an example. Hello Beeple!
/:cl:

### CODE REVIEW

**Comment by Crossedfall** [comment]
Review, but don't merge. I'll need to turn off the webhook before we do that.

**Comment by MCterra10** [comment]
based, oranges will steal this one, he's been wanting this for a while

File: tools/changelog/generate_cl.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        if v not in list(tags['defaults'].values()):
+            write_cl['changes'].append({tags['tags'][k]: v})
+
+if write_cl['changes']:
+    with open(Path.cwd().joinpath(f"html/changelogs/AutoChangeLog-pr-{pr_number}.yml"), 'w') as file:
+        yaml = yaml.YAML()
+        yaml.indent(sequence=4, offset=2)
+        yaml.dump(write_cl, file)
+
+    print(f"Done!")
```
Comment (qwertyquerty) [inline tools/changelog/generate_cl.py]: Why f

File: tools/changelog/generate_cl.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+pr_body = bytes(sys.argv[1], "utf-8").decode("unicode_escape")
+pr_number = sys.argv[2]
+pr_author = sys.argv[3]
+
+write_cl = {}
+try:
+    cl = CL_BODY.search(pr_body)
+    cl_list = CL_SPLIT.findall(cl.group(2))
+except AttributeError:
+    print("No CL!")
```
Comment (qwertyquerty) [inline tools/changelog/generate_cl.py]: Use your words

Comment (qwertyquerty) [inline tools/changelog/generate_cl.py]:
```suggestion
    print("No CL found!")
```

File: tools/changelog/generate_cl.py (hunk 1 above)
Comment (Crossedfall) [reply tools/changelog/generate_cl.py]: was going to include a var, decided not to.

**Comment by Crossedfall** [comment]
Bot is working, so this should be good to merge after checks.

**Comment by Crossedfall** [comment]
Edge-case re-run issue resolved in #1759

**Comment by Crossedfall** [comment]
Confirmed working:

https://github.com/BeeStation/BeeStation-Hornet/actions/runs/96720991
https://github.com/BeeStation/BeeStation-Hornet/actions/runs/96723640

**Pull request merged**